src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/MenuItemAutomationPeer.cs

```diff
@@ -195,12 +198,65 @@ protected override List<AutomationPeer> GetChildrenCore()
                                 children.Add(peer);
                         }
                     }
+
+                    return children;
                 }
             }
 
+            return GetChildrenFromVisualTree();
+        }
+
+        private List<AutomationPeer> GetChildrenFromVisualTree()
+        {
+            List<AutomationPeer> children = null;
+
+            iterate(Owner, ref children);
             return children;
         }
 
+        private static bool AddPeerToList(AutomationPeer peer, ref List<AutomationPeer> children)
+        {
+            if (children == null)
+                children = new List<AutomationPeer>();
+
+            children.Add(peer);
+            return false;
+        }
+
+        private static bool iterate(DependencyObject parent, ref List<AutomationPeer> children)
+        {
+            bool done = false;
+
+            if (parent != null)
+            {
+                AutomationPeer peer = null;
+                int count = VisualTreeHelper.GetChildrenCount(parent);
+                for (int i = 0; i < count && !done; i++)
+                {
+                    DependencyObject child = VisualTreeHelper.GetChild(parent, i);
+
+                    if (child != null
+                        && child is UIElement
+                        && (peer = CreatePeerForElement((UIElement)child)) != null)
+                    {
+                        done = AddPeerToList(peer, ref children);
+                    }
+                    else if (child != null
+                        && child is UIElement3D
+                        && (peer = UIElement3DAutomationPeer.CreatePeerForElement(((UIElement3D)child))) != null)
+                    {
+                        done = AddPeerToList(peer, ref children);
```

Comment:
The AddPeerToList always  return false. Can I ask when the `done` be false?

Reply:
I copied the method from UIElementAutomationPeer and eliminated the delegate. The "done" is a relict that's not required any more. I'll remove it.